makiflow/trainers/nn_render/render_trainer.py

```diff
@@ -0,0 +1,383 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# MakiFlow is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Foobar.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import
+import json
+import os
+import tensorflow as tf
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+import traceback
+import cv2
+import glob
+import copy
+
+from makiflow.models.nn_render.training_modules.abs_loss import ABS_LOSS
+from makiflow.models.nn_render.training_modules.mse_loss import MSE_LOSS
+from makiflow.models.nn_render.training_modules.masked_abs_loss import MASKED_ABS_LOSS
+from makiflow.models.nn_render.training_modules.masked_mse_loss import MASKED_MSE_LOSS
+from makiflow.models.nn_render.training_modules.perceptual_loss import PERCEPTUAL_LOSS
+
+
+from makiflow.trainers.utils.optimizer_builder import OptimizerBuilder
+from makiflow.tools.test_visualizer import TestVisualizer
+
+"""
+EXAMPLE OF THE TEST PARAMETERS:
+experiment_params = {
+    'name': name,
+    'epochs': 50,
+    'test period': 10,
+    'save period': None or int,
+    'optimizers': [
+        { 
+            type: ...
+            params: {}
+        }, 
+        { 
+            type: ...
+            params: {}
+        }
+    ]
+    'loss type': 'MakiLoss' or 'FocalLoss',
+    'batch sizes': [10],
+    'gammas': [2, 4],
+    'class names': [],
+    'weights': ../weights/weights.ckpt,
+    'path to arch': path,
+    'pretrained layers': [layer_name],
+    'utrainable layers': [layer_name],
+    'l1 reg': 1e-6 or None,
+    'l1 reg layers': [layer_name],
+    'l2 reg': 1e-6 or None,
+    'l2 reg layers': [layer_name]
+}"""
+
+
+class ExpField:
+    experiments = 'experiments'
+    name = 'name'
+    pretrained_layers = 'pretrained layers'
+    weights = 'weights'
+    untrainable_layers = 'untrainable layers'
+    epochs = 'epochs'
+    test_period = 'test period'
+    loss_type = 'loss type'
+    l1_reg = 'l1 reg'
+    l1_reg_layers = 'l1 reg layers'
+    l2_reg = 'l2 reg'
+    l2_reg_layers = 'l2 reg layers'
+    optimizers = 'optimizers'
+    iterations = 'iterations'
+    path_test_image = 'path_test_image'
+    path_test_uv = 'path_test_uv'
+    batch_size = 'batch_size'
+
+class SubExpField:
+    opt_info = 'opt_info'
+
+class LossType:
+    AbsLoss = 'AbsLoss'
+    MaskedAbsLoss = 'MaskedAbsLoss'
+    MaskedMseLoss = 'MaskedMseLoss'
+    MseLoss = 'MseLoss'
+    PerceptualLoss = 'PerceptualLoss'
+
+
+class RenderTrainer:
+    def __init__(self, model_creation_function, exp_params, path_to_save: str):
```

Comment:
Add docstring for each of the arguments. The API of the `model_creation_function` must be provided as well.